### DISCUSSION TITLE
Server-side model sync?

### DISCUSSION THREAD

**Discussion by boticello** [Jinntec/Fore#86] [Q&A] [answered]
I've started playing with Fore and like it very much. The Xforms model clicks with me, and your web components make it easy to work with.

How do I use a server-side model, such as an XML document in BaseX? Do you have plans to augment the model component with server connections, or do I need to set that up in some way?

**Comment by JoernT** [comment, marked as answer]
Hi,

not completely sure what you're referring to. However of course you can load and submit your data (aka model) from and to a server with the help of `instance/@src` and the `fx-submission` element. Of course your server needs to offer the appropriate API to serve the endpoints you're using on the client.

I'm mostly working with eXist-db as backend and it's very easy to connect Fore with some data model in the db - just load and store the data via a REST interface and that's it. You find examples in the demos dealing with submissions. Unfortunately i cannot put those demos online as they require a server. You should clone the code and run the demos locally to learn about the possibilities of fx-submission.

But i feel that there's some more behind your question. Can you elaborate a bit more?

Btw, i changed your question from an issue into a discussion - a rather new but cool feature of Github which is more appropriate for questions and discussions.

**Reply by boticello** [reply]
Thanks! Somehow I hadn't seen the submission demos, and so I hadn't realised fully how `fx-submission` worked. I can see the submission examples in the repo, and so I will try to connect them with eXist-db or BaseX as you indicate.

There's nothing more behind the question, really! Thanks. I'll come back to this discussion if I have any problems.
